File: storage/data_original/at_coder/abc030/B/4901574.py
# ??
n,m=map(int,input().split())

# ??????60??360?????1??6??
b=m*6

# ??????12???360?????1???30??
# 12????????0?11????????
a=n%12*30

# ???2??????????1?????
# ????????????
a+=m/2

# ????????????
print(min(abs(a-b),360-abs(a-b)))

File: storage/data_transpiled/at_coder/abc030/B/4901574.java
public static void print ( int n , int m ) {
  int b = m * 6 ;
  int a = n % 12 * 30 ;
  a += m / 2 ;
  System . out . println ( Math . min ( Math . abs ( a - b ) , 360 - Math . abs ( a - b ) ) ) ;
}
